### ISSUE TITLE
ESBDhex/gridify adds rows and coumns of non-existent pixels each run resulting in errors

### ISSUE THREAD

**Issue by kilir** [mtex-toolbox/mtex#676]
The bug is related to https://github.com/mtex-toolbox/mtex/discussions/631 and causes the number is non-indexed/ indexed pixels not to coincide with the number of pixels with a grain ID.

Steps to reproduce using Macros example data (https://github.com/mtex-toolbox/mtex/files/5769587/ebsd_test.zip)
```
load('ebsd_test.mat')
ebsd=ebsd_cut;

% adding the nonindexed  doesn't change the problem
ebsd = ebsd.gridify
% -> this gives 301 x 347
%idnon=isnan(ebsd.rotations);
%ebsd = EBSD(ebsd);
%ebsd(idnon).phaseId=1;

ebsd = ebsd.gridify
% -> this gives 302 x 347 ???

[grains,ebsd.grainId]=ebsd.calcGrains

% this triggers a subsequnet error (from EBSD/project2FundamentalRegion() called by KAM() )
isIndexed = ebsd.isIndexed;
refRot = ebsd.rotations(accumarray(ebsd.grainId(isIndexed),find(isIndexed),[],@max));

```

Rüdiger

**Comment by kilir** [comment]
It looks like the NaN phases which are added may be the culprit.
Adding 

`phaseId(isnan(phaseId))=1; `

in https://github.com/mtex-toolbox/mtex/blob/bf684df1b01b20a8ee397057eed328540b82bf71/EBSDAnalysis/%40EBSD/private/hexify.m#L135

seems to prevent the problem.

However, can we be sure that there's always a nonIndexed phase related to phaseId 1?

Rüdiger

**Comment by ralfHielscher** [comment]
Hi Rüdiger,

I hope the two commits fix the two issues raised here. #631 was actually an unrelated bug. which I fixed in 753b5f1.

Thank you for your work,

Ralf.

.
